--- service_upstart_linux.go	2022-05-30 17:46:47.843530283 +0800
+++ service_upstart_linux.fix.go	2022-06-01 14:05:42.269563760 +0800
@@ -242,10 +242,10 @@ stop on runlevel [!2345]
 {{if and .UserName .HasSetUIDStanza}}setuid {{.UserName}}{{end}}
 
 respawn
-respawn limit 10 5
+post-stop exec sleep 5
 umask 022
 
-console none
+console owner
 
 pre-start script
     test -x {{.Path}} || { stop; exit 0; }
@@ -264,6 +264,6 @@ script
 		set +a
 	fi
 
-	exec {{if and .UserName (not .HasSetUIDStanza)}}sudo -E -u {{.UserName}} {{end}}{{.Path}}{{range .Arguments}} {{.|cmd}}{{end}}{{if .LogOutput}} >> $stdout_log 2>> $stderr_log{{end}}
+	exec {{if and .UserName (not .HasSetUIDStanza)}}su {{.UserName}} -c {{end}}'{{.Path}}{{range .Arguments}} {{.|cmd}}{{end}}{{if .LogOutput}} >> $stdout_log 2>> $stderr_log{{end}}'
 end script
 `
